kythe/go/extractors/gcp/config/converter.go

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package config contains logic for converting
+// kythe.proto.extraction.RepoConfig to cloudbuild.yaml format as specified by
+// https://cloud.google.com/cloud-build/docs/build-config.
+package config
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	rpb "kythe.io/kythe/proto/repo_go_proto"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/protobuf/jsonpb"
+	cloudbuild "google.golang.org/api/cloudbuild/v1"
+)
+
+// Constants that map input/output substitutions.
+const (
+	corpus            = "${_CORPUS}"
+	outputFilePattern = "${_OUTPUT_KZIP_NAME}"
+	outputGsBucket    = "${_OUTPUT_GS_BUCKET}"
+	repoName          = "${_REPO_NAME}"
+)
+
+// Constants ephemeral to a single kythe cloudbuild run.
+const (
+	outputDirectory = "/workspace/out"
+	codeDirectory   = "/workspace/code"
+	javaVolumeName  = "kythe_extractors"
+)
+
+// KytheToYAML takes an input file, parses it as a JSON defined
+// kythe.proto.extraction.RepoConfig, and converts it to YAML.
+func KytheToYAML(input string) ([]byte, error) {
+	configProto, err := readConfigFile(input)
+	if err != nil {
+		return nil, fmt.Errorf("reading config file %s: %v", input, configProto)
+	}
+	build, err := KytheToBuild(configProto)
+	if err != nil {
+		return nil, fmt.Errorf("converting cloudbuild.Build: %v", err)
+	}
+	json, err := build.MarshalJSON()
+	if err != nil {
+		return nil, fmt.Errorf("marshalling cloudbuild.Build to JSON: %v", err)
+	}
+	return yaml.JSONToYAML(json)
+}
+
+// KytheToBuild takes a kythe.proto.extraction.RepoConfig and generates the
+// necessary cloudbuild.Build with BuildSteps for running kythe extraction.
+func KytheToBuild(conf rpb.Config) (cloudbuild.Build, error) {
+	build := cloudbuild.Build{}
+	initializeArtifacts(&build)
```

Comment:
I'm not go-savvy, but why create it just to initialize rather than having a function return a pointer to a new one?

Reply:
(already initialized)

Reply:
+1